packages/react-day-picker/src/contexts/Focus/FocusContext.tsx

```diff
@@ -41,11 +67,22 @@ export function FocusProvider({
   children: React.ReactNode;
 }): JSX.Element {
   const [focusedDay, setDay] = React.useState<Date | undefined>();
-  const { goToMonth, displayMonths } = useNavigation();
+  const [tempoFocusedDay, setTempo] = React.useState<Date | undefined>();
+  const initialFocusChanged = React.useRef(false);
```

Comment:
Why do we need to track the initial focus? I wasn't expecting that...

```suggestion
```

Reply:
The initial focus changes every time the user navigates with arrow keys or selects a specific day.  
In order for navigation by months and years to focus on the same date as the initial focus (before PageUp, PageDowm, and other related keys were pressed) it needs to separate the initial focus state and the dynamic focus states. Meaning, when the user presses PageUp several times, the calendar will go months back and still focus on the **same** date that was saved before any of the extended keyboard navigation keys were pressed.